src/decorators/relationships.js

```diff
@@ -185,8 +185,10 @@ export function setRelatedRecord (record, relatedRecord, property, modelType = n
 
   if (!relatedRecord) {
     delete relationships[relationType]
+    _dirtyRelationships.add(relationType)
   } else if (!data || !(data.type === type && data.id === id)) {
     relationships[relationType] = { data: { id, type } }
+    _dirtyRelationships.add(relationType)
```

Comment:
this case covers when the relatedRecord is changed, yes? There isn't a test for that use case but the attribute should be marked as dirty.

Reply:
this is for creating a relationship. The other is for updating it.